__tests__/account.test.ts

```diff
@@ -501,18 +509,56 @@ describe('deploy and test Wallet', () => {
 describeIfDevnetSequencer('not implemented for RPC', () => {
   // Testnet will not accept declare v2 with same compiledClassHash,
   // aka. we can't redeclare same contract
-  describe('Test Cairo 1', () => {
-    const provider = getTestProvider();
+  describe('Cairo 1', () => {
+    const provider = getTestProvider() as SequencerProvider;
     const account = getTestAccount(provider);
+    let classHash: any;
+    let contractAddress: any;
+    let declareV2Tx: any;
     initializeMatcher(expect);
 
+    beforeAll(async () => {
+      declareV2Tx = await account.declare({
+        contract: compiledHelloSierra,
+        casm: compiledHelloSierraCasm,
+      });
+      classHash = declareV2Tx.class_hash;
+      await provider.waitForTransaction(declareV2Tx.transaction_hash);
+      const { transaction_hash, contract_address } = await account.deploy({ classHash });
+      [contractAddress] = contract_address;
+      await provider.waitForTransaction(transaction_hash);
+    });
+
     test('Declare v2 - Hello Cairo 1 contract', async () => {
-      const declareTx = await account.declare({
+      expect(declareV2Tx).toMatchSchemaRef('DeclareContractResponse');
+    });
+
+    xtest('declareAndDeploy Cairo 1', async () => {
+      // Can't test due to redeployment not allowed
+      await account.declareAndDeploy({
         contract: compiledHelloSierra,
         casm: compiledHelloSierraCasm,
       });
-      await provider.waitForTransaction(declareTx.transaction_hash);
-      expect(declareTx).toMatchSchemaRef('DeclareContractResponse');
+    });
+
+    test('getCompiledClassByClassHash', async () => {
+      const compiledClass = await provider.getCompiledClassByClassHash(classHash);
+      expect(compiledClass).toMatchSchemaRef('CompiledClass');
+    });
+
+    test('GetClassByHash', async () => {
+      const classResponse = await provider.getClassByHash(classHash);
+      expect(classResponse).toMatchSchemaRef('SierraContractClass');
+    });
+
+    test('GetClassAt', async () => {
+      const classResponse = await provider.getClassAt(contractAddress);
+      expect(classResponse).toMatchSchemaRef('SierraContractClass');
+    });
+
+    test('getCompiledClassByClassHash', async () => {
```

Comment:
Is this a duplicate of the third preceding/last test (the referenced schema as well)?

Nitpick: The `GetClassByHash` and `GetClassAt` tests could be changed to use `account` in place of `provider` so they align closer to the test file name.

Reply:
Yea,
I think I will extract this all to the new cairo1 test where we will update also contract interactions